pandas/_libs/tslibs/period.pyx

```diff
@@ -1186,6 +1186,13 @@ cdef int64_t period_ordinal_to_dt64(int64_t ordinal, int freq) except? -1:
     if ordinal == NPY_NAT:
         return NPY_NAT
 
+    if freq == 11000:
+        # Microsecond, avoid get_date_info to prevent floating point errors
+        pandas_datetime_to_datetimestruct(ordinal, NPY_FR_us, &dts)
```

Comment:
unrelated but have you ever thought about renaming this function to something like "int64ns_to_datetimestruct"? I always find myself thinking about what a "pandas_datetime" is

Reply:
i generally agree.  i think it got this name because the numpy version is numpy_datetime_to_datetimestruct or something along those lines.